_posts/2016-10-16-getting-started-in-scala-on-android.md

```diff
@@ -0,0 +1,217 @@
+
+##Required tools
+In order to develop Android apps in Scala, you need a minimum set of tools: [SBT](http://www.scala-sbt.org/) and [Android SDK](https://developer.android.com/studio/index.html).
+
+###Install SBT
+You can install SBT on Mac OSX using [brew](http://brew.sh/).
+
+```bash
+$ brew install sbt
+```
+
+To install SBT on other operating systems, you can follow the [instructions](http://www.scala-sbt.org/0.13/docs/Setup.html) on the official documentation.
+
+###Install the Android SDK
+You can just download the latest version of the Android SDK from the [Developers website](https://developer.android.com/studio/index.html) and follow the installation instructions.
+Alternatively, you can install [Android Studio](https://developer.android.com/studio/index.html), which comes with the Android SDK and emulators.
+
+###Set the ANDROID_HOME environment variable
+On Mac OSX/Linux, you can just export the variable
+
+```bash
+$ export ANDROID_HOME=path_to_your_android_sdk
+```
+
+or add it to your bash_rc or bash_profile.
+
+NOTE: On Mac OSX, if Android Studio is installed, the Android SDK is usually located at `/Users/your_user_name/Library/Android/sdk/`
+
+##Add SBT plugin
+The easiest way to install the Android SDK Plugin for SBT is to do it globally. For this, you'll need to create a file in the SBT plugins folder:
+
+```bash
+~/.sbt/0.13/plugins/android.sbt
+```
+
+and add the following line:
+
+```bash
+addSbtPlugin("org.scala-android" % "sbt-android" % "1.7.0")
+```
+
+##Create the Android project
+Let's create a folder for our Android project:
+
+```bash
+$ mkdir my-project
+$ cd my-project
+```
+
+Now, we are going to use the SBT plugin to create a template project. First, run SBT:
+
+```bash
+$ sbt
+```
+
+Then, use the plugin to create the project.
+
+```bash
+> gen-android <package_name> <project_name>
+```
+
+For example:
+
+```bash
+> gen-android com.codurance scala_on_android
+```
+
+The project structure created looks like this:
+
+```text
+my-project/
+|-- project/
+|   |-- android.sbt
+|   |-- build.properties
+|-- src/
+|   |-- androidTest/
+|       |-- java/
+|           |-- com/
+|               |-- codurance/
+|                   |-- Junit3MainActivityTest.java
+|                   |-- Junit4MainActivityTest.java
+|   |-- main/
+|       |-- res/
+|           // Android resorces folders
+|       |-- scala/
+|           |-- com/
+|               |-- codurance/
+|                   |-- MainActivity.scala
+|       |-- AndroidManifest.xml
+|-- build.sbt
+|-- lint.xml
+```
+
+The SBT plugin creates a project structure with the minimum files needed to run an Android project, plus a setup for running instrumentation tests. Notice that the test classes generated are in Java, and the Main Activity is in Scala.
+
+The most interesting file is build.sbt. I've added some comments to explain what's the purpose of each line.
+
+```java
```

Comment:
I tried Scala, but the highlighting was a bit weird and less readable, so I switched to Java. I wasn't particularly happy about this, but was more readable. I attach an image below to compare:
<img width="705" alt="scala-formatting" src="https://cloud.githubusercontent.com/assets/5873601/19437146/f45ce572-946b-11e6-8316-1481411d8974.png">